### PR TITLE
remove duplicate code related to ccache loadfile

### PR DESCRIPTION
**Pull request by rmaksimov**
I've added the parseFile function to ccache.py and have refactored all the duplicate code in other scripts

### CODE REVIEW

File: impacket/krb5/ccache.py [hunk 1]
```diff
@@ -520,16 +521,65 @@ def fromTGS(self, tgs, oldSessionKey, sessionKey):
...
+            raise e
 
     def saveFile(self, fileName):
-        f = open(fileName,'wb+')
+        f = open(fileName, 'wb+')
         f.write(self.getData())
         f.close()
 
+    @classmethod
+    def parseFile(cls, domain, username='', target=''):
```
Comment (martingalloar) [inline impacket/krb5/ccache.py:542]: Maybe we can add some docstrings to make sure it's clear what's expected for each parameter.

File: impacket/tds.py [hunk 2]
```diff
@@ -698,71 +698,32 @@ def kerberosLogin(self, database, username, password='', domain='', hashes=None,
...
-
-                # retrieve user information from CCache file if needed
-                if username == '' and creds is not None:
-                    username = creds['client'].prettyPrint().split(b'@')[0].decode('utf-8')
-                    LOG.debug('Username retrieved from CCache: %s' % username)
-                elif username == '' and len(ccache.principal.components) > 0:
-                    username = ccache.principal.components[0]['data'].decode('utf-8')
-                    LOG.debug('Username retrieved from CCache: %s' % username)
+        if useCache:
+            domain, username, _, TGS = CCache.parseFile(domain, username, 'MSSQLSvc/%s:%d' % (self.server, self.port))
```
Comment (martingalloar) [inline impacket/tds.py:711]: @0xdeaddood pointed out that here obtaining the TGT is required because it will be later on needed.

**Review by martingalloar — CHANGES_REQUESTED** [review]

**Comment by rmaksimov** [comment]
thanks for the review, i seem to have fixed all the requested things

**Comment by 0xdeaddood** [comment]
thanks a lot for this PR @rmaksimov!

**Pull request merged**